python/GafferImageUI/ImageInspector.py

```diff
@@ -0,0 +1,460 @@
+			_maxNodes = 1 if self.__settingsNode["in"].getInput() is not None else 0,
+			_reverseNodes = True,
+			_ellipsis = False
+		)
+
+	def __plugParentChanged( self, plug, oldParent ) :
+
+		# The plug we were viewing has been deleted or moved - find
+		# another one to view.
+		self._updateFromSet()
+
+	@GafferUI.LazyMethod( deferUntilPlaybackStops = True )
+	def __setPathListingPaths( self ) :
+
+		# We take a static copy of our current context for use in the paths. This
+		# lets us defer updates until playback stops, and also provides important
+		# thread-safety, because the PathListingWidget will access the context
+		# on a background thread.
+		context = Gaffer.Context( self.getContext() )
+		self.__formatPathListing.setPath(
+			_FormatPath( self.scriptNode(), self.__settingsNode["__selectView"]["out"], context, "/" ),
+		)
+		self.__channelsPathListing.setPath(
+			_ChannelPath( self.scriptNode(), self.__settingsNode["__imageStats"]["in"], context, "/" ),
+		)
+		self.__metadataPathListing.setPath(
+			_MetadataPath( self.scriptNode(), self.__settingsNode["__selectView"]["out"], context, "/" ),
+		)
+
+GafferUI.Editor.registerType( "ImageInspector", ImageInspector )
+
+##########################################################################
+# Metadata controlling the settings UI
+##########################################################################
+
+Gaffer.Metadata.registerNode(
+
+	ImageInspector.Settings,
+
+	plugs = {
+
+		"view" : [
+
+			"plugValueWidget:type", "GafferImageUI.ViewPlugValueWidget",
+			"description", "The view to inspect",
+
+		],
+
+	}
+
+)
+
+##########################################################################
+# _ImagePath
+##########################################################################
+
+class _ImagePath( Gaffer.Path ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		Gaffer.Path.__init__( self, path, root, filter )
+
+		self.__scriptNode = scriptNode
+		self._context = context
+		self._image = image
+		self.__imagePlugDirtiedConnection = self._image.node().plugDirtiedSignal().connect(
+			Gaffer.WeakMethod( self.__imagePlugDirtied, fallbackResult = None ), scoped = True
+		)
+
+	def copy( self ) :
+
+		return self.__class__( self.__scriptNode, self._image, self._context, self[:], self.root(), self.getFilter() )
+
+	def cancellationSubject( self ) :
+
+		## \todo We really just want to return `self._image` here, but we
+		# can't because BackgroundTask can't find the right ScriptNode from it.
+		return self.__scriptNode["fileName"]
+
+	def _children( self, canceller ) :
+
+		return [
+			self.__class__( self.__scriptNode, self._image, self._context, self[:] + [ name ], self.root(), self.getFilter() )
+			for name in self._childNames( canceller )
+		]
+
+	def _childNames( self, canceller ) :
+
+		raise NotImplementedError
+
+	def __imagePlugDirtied( self, plug ) :
+
+		if plug.parent() == self._image and self._imagePlugAffectsPath( plug ) :
```

Comment:
Do you think it would be worth adding a definition for `_imagePlugAffectsPath` to this base class that raise `NotImplementedError` just like for the `_childNames`?

Reply:
Good idea - done in https://github.com/GafferHQ/gaffer/pull/5598/commits/0dc621edb840e79b0b61e31a6e244f0539512593.